ui/pages/notifications/notification-components/snap/snap.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { NotificationListItem } from '../../../../components/multichain';
+import type { SnapNotificationWithoutSnapName } from '../../snap/types/types';
+import { NotificationListItemIconType } from '../../../../components/multichain/notification-list-item-icon/notification-list-item-icon';
+import { getSnapsMetadata } from '../../../../selectors';
+import { markNotificationsAsRead } from '../../../../store/actions';
+import { getSnapRoute, getSnapName } from '../../../../helpers/utils/util';
+import { TextVariant } from '../../../../helpers/constants/design-system';
+
+type SnapComponentProps = {
+  snapNotification: SnapNotificationWithoutSnapName;
+};
+
+export const SnapComponent = ({ snapNotification }: SnapComponentProps) => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const snapsMetadata = useSelector(getSnapsMetadata);
+
+  const snapsNameGetter = getSnapName(snapsMetadata);
+
+  const handleSnapClick = () => {
+    dispatch(markNotificationsAsRead([snapNotification.id]));
+    history.push(getSnapRoute(snapNotification.data.origin));
+  };
+
+  return (
+    <NotificationListItem
+      id={snapNotification.id}
+      isRead={snapNotification.isRead}
+      createdAt={new Date(snapNotification.createdAt)}
+      title={{
+        items: [
+          {
+            text: snapsNameGetter(snapNotification.data.origin) || 'Snap',
+          },
+        ],
+      }}
+      description={{
+        items: [
+          {
+            text: snapNotification.data.message,
+          },
+        ],
+        variant: TextVariant.bodyMd,
+      }}
+      icon={{
+        type: NotificationListItemIconType.Token,
+        value: 'https://s2.coinmarketcap.com/static/img/coins/64x64/13855.png',
```

Comment:
Why do we have the ENS token icon as the snap notification icon ?

Reply:
I agree.

<details><summary>See Original Markup</summary>

```jsx
    <div className="notifications__item" onClick={handleItemClick}>
      <div
        className={classnames(
          'notifications__item__unread-dot',
          !readDate && 'unread',
        )}
      />
      <div className="notifications__item__details">
        <div className="notifications__item__details__message">
          <SnapUIMarkdown markdown>{message}</SnapUIMarkdown>
        </div>
        <p className="notifications__item__details__infos">
          {t('notificationsInfos', [
            formatDate(createdDate, "LLLL d',' yyyy 'at' t"),
            <Button type="inline" onClick={handleNameClick} key="button">
              {snapName}
            </Button>,
          ])}
        </p>
      </div>
    </div>
```
</details> 

The questions that come to mind are:
1. Why are we hardcoding this image?
2. Why are we not using the markdown component?
3. _(high level)_, why are we trying to parse a Snap Notification Component to fit our general Notification shape?


I understand we want to have a cohesive notification experience - [see figma](https://www.figma.com/design/c7GgNw2kScGrVyRGAPhwEd/Notifications?node-id=1030-98323&t=SOTESLbmdFjCOoqU-0) - but lets be realistic. Snaps (and snap markup) is very flexible, I don't think we should try parsing it into our shapes/components.

Instead if we see snap notifications, we can just use the original renderer/components.
- We may need to tweak the original a little (e.g. width sizing or padding), but should it majority untouched

Reply:
@matteoscurati ☝🏾

Reply:
To keep this unblocked lets revert the snap rendering & styling to the original. We can sync up the teams to discuss the new UI if we want some compatability.

E.g. example of reverting snaps UI with some new UI below (some feature announcements)

<img width="1088" alt="Screenshot 2024-05-23 at 01 47 43" src="https://github.com/MetaMask/metamask-extension/assets/17534261/0c869637-4d03-40ae-a3eb-0ac064efd4d2">

Reply:
Allowing remote-loading of images sounds like a security issue. Any images known to be needed need can be bundled into (the snap / extension) at build-time.

Reply:
Thank you @Prithpal-Sooriya . Actually, we discussed this topic months ago, evaluating how to adapt Snap notifications to the new notifications designed and approved in Figma. The markdown topic wasn't mentioned at that time. The use of the image was also discussed, and it was agreed, at least for now, to have a single image for all Snap notifications.

@FrederikBolding @GuillaumeRx

Reply:
Instead of using the old component, can’t we just add this part inside the new component?

<img width="538" alt="Screenshot 2024-05-23 at 05 33 41" src="https://github.com/MetaMask/metamask-extension/assets/1284304/0e644f8e-549c-40d6-8ca8-ea355c9f3f54">

Reply:
Em, if it is possible try it. Otherwise just do a full revert.
@legobeat I 100% agree! We are removing and reverting snap notifications to the original.